### ISSUE TITLE
egl example mmap error

### ISSUE THREAD

**Issue by madushan1000** [MaxVerevkin/wayrs#9]
I'm trying to get the egl example working. when I run it I get an mmap permissions error, I'm running the latest code from the main branch(commit c7b03ab4b78001ef59774de821c16de1ab397f1a) on Ubuntu 23.10. I couldn't figure out why it would fail.

```
❯ cargo run --example triangle
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/examples/triangle`
thread 'main' panicked at /home/madushan/Dev/my/rust-deps/wayrs-orig/wayrs-utils/src/dmabuf_feedback.rs:129:22:
mmap failed: Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by MaxVerevkin** [comment]
Interesting. Which compositor are you using?

**Comment by madushan1000** [comment]
mutter, I'm using gnome 45.2. I will try some other compositor too.

**Comment by MaxVerevkin** [comment]
Oh, I have an idea why that may be happening. Can you try patching the line which panics to use `map_copy_read_only` to instead of `map`?

**Comment by madushan1000** [comment]
that works! thanks :)

**Comment by MaxVerevkin** [comment]
So it should be fixed in c42362e

**Comment by madushan1000** [comment]
nice! thanks :)
